pkg/autoscaler/statserver/server.go

```diff
@@ -113,13 +123,28 @@ func (s *Server) Handler(w http.ResponseWriter, r *http.Request) {
 	if handleHealthz(w, r) {
 		return
 	}
+
 	var upgrader websocket.Upgrader
 	conn, err := upgrader.Upgrade(w, r, nil)
 	if err != nil {
 		s.logger.Errorw("error upgrading websocket", zap.Error(err))
 		return
 	}
 
+	if s.isBktOwner != nil && isBucketHost(r.Host) {
```

Comment:
One last question: How's the flow of this code supposed to be triggered? :thinking: Why are we even generating the connection above if we close it immediately?

Reply:
If there's mesh, one autoscaler connects another autoscaler with host like `autoscaler-bucket-00-of-10`. If the target pod is not the owner of that bucket (it happens when the ownership changes from one pod to another while the old pod is still healthy and there's delay for the endpoints change to propagate), it closes the connection.

I thought closing the connection will trigger a re-connection for the client (managedConnection lib). Will just return before setting up a connection achieve the reconnection as well? (will have a try)

Reply:
Verified we can just return without creating the connection. Updated.

Reply:
Prototype is like: https://github.com/yanweiguo/serving/blob/websocket_svc/cmd/autoscaler/main.go#L194